### PR TITLE
Implement Try It Button

### PR DESCRIPTION
**Pull request by blangley28**
Try it button now redirects to the /demo endpoint which shows a banner explaining nothing will be saved. The banner links to the login and signup modals which have been refactored to allows them to be easily used in the two different components.

### CODE REVIEW

**Comment by danlee1025** [comment]
This looks pretty good, just one question. Is clicking on the the camino logo supposed to bring us back to the landing page?

File: src/components/StaticApp/Header/Header.tsx [hunk 1]
```diff
@@ -10,6 +10,29 @@ import { useThemeToggle } from 'components/ThemeProvider';
...
+
+  return (
+    <div className={styles.root}>
+      <FontAwesomeIcon icon={faExclamationCircle} className={styles.icon} />
+      You are currently viewing syntheic demo data. Any changes made will not be saved. To create
+      your own pathways,{' '}
+      <span className={styles.underline} onClick={openLoginModal}>
+        login
+      </span>{' '}
+      or{' '}
```
Comment (radamson) [inline src/components/StaticApp/Header/Header.tsx:28]: Does the space have to be in `{' '}`?  Maybe its doing something I don't understand.

File: src/index.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ import ThemeProvider from 'components/ThemeProvider';
 
 import './styles/index.scss';
 import StaticApp from 'components/StaticApp';
+import { AuthModalProvider } from 'components/Auth/AuthModalProvider';
```
Comment (radamson) [inline src/index.js:14]: any reason `AuthModalProvider` export isn't added to `Auth/index.ts`?  I don't have a strong opinion on it, but some of the components in the directory are included and some aren't

File: src/components/StaticApp/Header/Header.tsx (hunk 1 above)
Comment (blangley28) [reply src/components/StaticApp/Header/Header.tsx:28]: Everytime I deleted it the formatter just added it back. I removed it in a different editor and replaced them with spaces but the spaces don't render - so I guess it is required.
Screen Shot 2020-11-12 at 2 12 38 PM

File: src/index.js (hunk 2 above)
Comment (blangley28) [reply src/index.js]: Nope - added to `Auth/index.ts`

**Review by radamson — APPROVED** [review]
Tested the `/demo` and `/builder` pages and they both were working as expected.  If you switch to another page via the browser URL (e.g. `/demo` to `/builder`) with either the sign-up or login modals open they stay open, but you can still close them so its not a big deal

**Comment by blangley28** [comment]
> Is clicking on the the camino logo supposed to bring us back to the landing page?

Talked with Reece and we decided clicking the logo on the static app should bring the user back to the landing page but not from the backend builder. Updated with this functionality.

**Review by danlee1025 — APPROVED** [review]

**Pull request merged**